### PR TITLE
Delete experimental table from playground

### PR DESCRIPTION
**Pull request by Fetz**
Deleting the experimental table from the playground for the following reasons:
- The experimental table was not kept up to date with the non version
- Having 2 tables it only confuse new users
- Create noise (issues being raised for experimental table etc)
- If anyone is dependent on the experimental table they can still create a plugin based on the deleted code from this Pull Request or check the git history

Manual testing the playground and creating a table:
https://github.com/facebook/lexical/assets/1071113/92212a2d-0047-4915-8e01-3c7fb57849ac

### CODE REVIEW

File: packages/lexical-playground/src/plugins/ComponentPickerPlugin/index.tsx [hunk 1]
```diff
@@ -171,22 +170,6 @@ function getBaseOptions(editor: LexicalEditor, showModal: ShowModal) {
             }),
         }),
     ),
-    new ComponentPickerOption('Table', {
```
Comment (ivailop7) [inline packages/lexical-playground/src/plugins/ComponentPickerPlugin/index.tsx:174]: we still want to keep the standard table one

**Review by ivailop7 — COMMENTED** [review]
This will need indeed need some untangling.

**Comment by ivailop7** [comment]
@Fetz deployment is still failing, let me know when it's ready for review.

**Comment by GermanJablo** [comment]
I would greatly appreciate if you could provide context about this PR.

I opened a thread on discord about this:
https://discord.com/channels/953974421008293909/1177620481533284463

**Comment by Fetz** [comment]
@ivailop7 sorry for the delay, fixed the deployment and tested the playground.

**Review by ivailop7 — APPROVED** [review]
LGTM

**Comment by agriffis** [comment]
This missed `packages/lexical-playground/src/nodes/TableNode.tsx` which should also be deleted.

**Comment by agriffis** [comment]
I just added the removal of `TableNode.tsx` to #5342 because otherwise `tsc` fails

**Pull request merged**